ipn/ipnlocal/local.go

```diff
@@ -3795,55 +3795,116 @@ func (b *LocalBackend) shouldUploadServices() bool {
 //
 // On non-multi-user systems, the actor should be set to nil.
 func (b *LocalBackend) SetCurrentUser(actor ipnauth.Actor) {
-	var uid ipn.WindowsUserID
-	if actor != nil {
-		uid = actor.UserID()
-	}
-
 	unlock := b.lockAndGetUnlock()
 	defer unlock()
 
+	var userIdentifier string
+	if user := cmp.Or(actor, b.currentUser); user != nil {
+		maybeUsername, _ := user.Username()
+		userIdentifier = cmp.Or(maybeUsername, string(user.UserID()))
+	}
+
 	if actor != b.currentUser {
 		if c, ok := b.currentUser.(ipnauth.ActorCloser); ok {
 			c.Close()
 		}
 		b.currentUser = actor
 	}
 
-	if b.pm.CurrentUserID() == uid {
-		return
-	}
-
-	var profileID ipn.ProfileID
+	var action string
 	if actor != nil {
```

Comment:
Since this is now a simple `if`/`else`, simplify by reversing the predicate to `actor == nil` and swapping the blocks? I think it could improve readability a bit.